src/principal-vs-secondary/regis/Diagonal.java

```diff
@@ -0,0 +1,66 @@
+import java.util.Random;
+public class Diagonal {
+
+    public int [][] matrixRandom(int side){
+        int [][] matrix = new int [side][side];
+        Random numRand =  new Random();
+        for (int x = 0; x < side; x++) {
+            for (int y = 0; y < side; y++) {
+                matrix[x][y] = numRand.nextInt(9) + 1;
+            }
+        }
+        return matrix;
+    }
+
+    public int addDiagonalMain(int [][] matrix){
```

Comment:
This method name could be a little bit confusing. If you use it from an instance of this class, it will look like `diagonal.addDiagonalMain(matrix)`, which seems to say `Add a diagonal matrix to the diagonal instance`